pkg/bpf/bpf.go

```diff
@@ -305,6 +314,10 @@ func OpenOrCreateMap(path string, mapType int, keySize, valueSize, maxEntries ui
 		if err != nil {
 			return 0, isNewMap, err
 		}
+
+		// Need to close this fd since the next ObjGet
+		// returns a different fd.
+		ObjClose(fd)
```

Comment:
>we need to close the prior created fd in ObjPin()'s error path.

you mean the one created by `C.create_bpf_obj_pin(C.int(fd), pathStr, unsafe.Pointer(&uba))`?

>Why not just return the fd at that point, since we already have one?

@tgraf can respond to that better than me